### PR TITLE
ml.mdwalters.EightBall

### PR DESCRIPTION
**Pull request by mdwalters**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
this PR is supposed to be against `new-pr`

File: ml.mdwalters.EightBall.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "eight-ball",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "file:///home/ubuntu/Projects/eight-ball"
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.json:33]: it will definitely not find the source code.

File: ml.mdwalters.EightBall.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                }
+            ]
+        }
+    ],
+    "build-options" : {
+        "env" : {        }
+    }
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.json:40]:
```suggestion
    ]
```

File: ml.mdwalters.EightBall.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<component type="desktop">
+	<id>ml.mdwalters.EightBall.desktop</id>
+	<metadata_license>CC0-1.0</metadata_license>
+	<project_license>MIT</project_license>
+  <name>Eight Ball</name>
+  <summary>Make decisions easily</summary>
+	<description>
+	  <p>Eight Ball makes it more easier to make decisions, on whatever it is, from deciding if doing something is a good idea, or it's a bad idea!</p>
+	</description>
+</component>
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.appdata.xml:11]: this is insufficient

File: ml.mdwalters.EightBall.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+	<id>ml.mdwalters.EightBall.desktop</id>
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.appdata.xml:3]:
```suggestion
	<id>ml.mdwalters.EightBall</id>
```

File: ml.mdwalters.EightBall.appdata.xml (hunk 3 above)
Comment (hfiguiere) [reply ml.mdwalters.EightBall.appdata.xml:11]: this is not resolved.

File: ml.mdwalters.EightBall.json (hunk 1 above)
Comment (hfiguiere) [reply ml.mdwalters.EightBall.json:33]: this is not resolved.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: ml.mdwalters.EightBall.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "eight-ball",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/mdwalters/eight-ball"
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.json:33]: you need a revision or tag here.

File: ml.mdwalters.EightBall.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+	<id>ml.mdwalters.EightBall</id>
+	<metadata_license>CC0-1.0</metadata_license>
+	<project_license>MIT</project_license>
+  <name>Eight Ball</name>
+  <summary>Make decisions easily</summary>
+	<description>
+	  <p>Eight Ball makes it more easier to make decisions, on whatever it is, from deciding if doing something is a good idea, or it's a bad idea!</p>
+	</description>
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.appdata.xml:10]: as I previously said, this appstream file not sufficient...

missing `releases`, `screenshots`, `url`, `content_rating`

File: ml.mdwalters.EightBall.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                }
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.json:34]: upstream repository already containt desktop file, appstream and icon. They are not necessary in the repository.

**Comment by hfiguiere** [comment]
bot, build ml.mdwalters.EightBall

File: ml.mdwalters.EightBall.json (hunk 5 above)
Comment (mdwalters) [reply ml.mdwalters.EightBall.json:33]: how would i add that?

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by mdwalters** [comment]
why won't it work?

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/ml.mdwalters.EightBall.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
• tag-missing           : <url> is not present
Validation of files failed
```

File: ml.mdwalters.EightBall.appdata.xml (hunk 6 above)
Comment (hfiguiere) [reply ml.mdwalters.EightBall.appdata.xml:10]: still not done

File: ml.mdwalters.EightBall.desktop [hunk 8]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.desktop:1]: this is upstream too, remove this from here.

File: ml.mdwalters.EightBall.appdata.xml (hunk 6 above)
Comment (hfiguiere) [reply ml.mdwalters.EightBall.appdata.xml:10]: but it should actually be removed from here it's upstream.

File: ml.mdwalters.EightBall.json [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+            "name" : "eight-ball",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "archive",
+                    "url" : "https://github.com/mdwalters/eight-ball/archive/refs/tags/1.3.2.zip",
+                    "sha256": "e34f9ece7b1dedc83048a4c06bbcbac859ecce16eae7f165ffeb2f406f956ee3",
+                    "x-checker-data" : {
+                        "type" : "gnome",
```
Comment (hfiguiere) [inline ml.mdwalters.EightBall.json:36]: I doubt his is valid since the tarball isn't hosted on gnome.

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by hfiguiere** [comment]
it will not magically fix by itself. I already commented one what needed to be done. Either a patch or upstream.

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by hfiguiere** [comment]
```
ml.mdwalters.EightBall:            AppData problem: tag-invalid : Expected children for tag
ml.mdwalters.EightBall:            AppData problem: tag-invalid : <url> type invalid [unknown]
```

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by mdwalters** [comment]
uh...

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/ml.mdwalters.EightBall.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• attribute-missing     : <release> has no timestamp
Validation of files failed
```

**Comment by barthalion** [comment]
bot, build ml.mdwalters.EightBall

**Comment by hfiguiere** [comment]
stll happening.
```
builddir/files/share/appdata/ml.mdwalters.EightBall.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• attribute-missing     : <release> has no timestamp
Validation of files failed
```

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by mdwalters** [comment]
i think this time it'll work

**Comment by mdwalters** [comment]
bot, build ml.mdwalters.EightBall

**Comment by mdwalters** [comment]
YES!

**Pull request closed without merging**